src/main/java/com/google/starfish/models/Note.java

```diff
@@ -0,0 +1,55 @@
+package com.google.starfish.models;
+
+import java.util.Date;
+
+/** Represents a note */ 
+public class Note {
+
+  private long id;
+  private long authorId;
+  private String school;
+  private String course;
+  private String title;
+  private String sourceUrl;
+  private String pdfSource;
+  private Date dateCreated;
+  private long numDownloads;
+  private long numFavorites;
+  private String[] miscLabels;
+
+  /** Default constructor that throws null exception if NOT NULL fields are null */
+  public Note(long id, 
+      long authorId, 
+      String school, 
+      String course, 
+      String title, 
+      String sourceUrl, 
+      String pdfSource, 
+      Date dateCreated, 
+      long numDownloads, 
+      long numFavorites, 
+      String[] miscLabels) {
+
+    if (id == null || 
```

Comment:
Instead of doing this, could you declare some of the parameters as non-nullable? (Less familiar with Java so genuinely asking)

Reply:
I'm not as familiar with Java either but I just did some research into it and it seems like you can use ```Objects.requireNonNull()```. [Here's the SO Post](https://stackoverflow.com/questions/45632920/why-should-one-use-objects-requirenonnull) I'm referencing. I'll push changes to use this for my parameter Null checking instead of these long if statements because it seems a lot more readable.

Reply:
Nice looks more readable.